components/file_system/src/iosnoop/biosnoop.rs

```diff
@@ -0,0 +1,332 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::metrics::*;
+use super::IOStats;
+use crate::IOType;
+
+use collections::HashMap;
+use std::collections::VecDeque;
+use std::ffi::CString;
+use std::ptr;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Mutex;
+
+use bcc::{table::Table, Kprobe, BPF};
+use crossbeam_utils::CachePadded;
+
+const MAX_THREAD_IDX: usize = 128;
```

Comment:
Is this enough? How many threads TiKV spawns normally?

Reply:
unified pool(0.8*LOGIC_CPU_NUM as most), grpc(6), scheduler-worker-pool(4), raftstore(3), apply pool(3), rocksdb(10) with some task worker which would not exceed 10.

Reply:
to be on the safe side, let's make it to 192